lib/modules/hosts/xkcd.js

```diff
@@ -0,0 +1,28 @@
+addLibrary('mediaHosts', 'xkcd', {
+	domains: ['xkcd.com'],
+	logo: 'https://xkcd.com/favicon.ico',
+	detect: href => (/^https?:\/\/(?:www\.|m\.)?xkcd\.com\/([0-9]+)/i).exec(href),
```

Comment:
Unrelated to this PR, but it would be neat if [showImages `detect` call](https://github.com/thybag/Reddit-Enhancement-Suite/blob/feature/xkcd-alt/lib/modules/showImages.js#L613) would `detect.test ? detect.test(href) : detect.call ? mod.detect(elem.href, elem) : false`